recurring_contract/models/move.py

```diff
@@ -178,8 +177,101 @@ def find_sum(numbers, target, partial=None):
                     if payment_line.credit > missing_amount:
                         # Split last added line amount to perfectly match
                         # the total amount we are looking for
-                        return (open_payments[:index + 1] | move_lines)\
+                        return (open_payments[:index + 1] | move_lines) \
                             .split_payment_and_reconcile()
                     payment_amount += payment_line.credit
                 return (open_payments | move_lines).reconcile()
 
+    def update_invoices(self, updt_val):
+        """
+        It updates the invoices in self with the value of updt_val
+
+        :param updt_val: a dictionary of invoices values with the invoice name
+        which refer to another dictionnary of values for that invoice name
+        """
+        for invoice in self:
+            invoice.button_draft()
+            # Retrieve the value for a specific invoice
+            val_to_updt = updt_val.get(invoice.name)
+            # If the invoice is defined to_cancel we update it state
+            if "to_cancel" in val_to_updt["invoice_line_ids"]:
+                invoice._cancel_invoice()
+                continue
+            else:
+                invoice.write(val_to_updt)
+            invoice.action_post()
```

Comment:
Based on my other change proposition, the dictionary could be empty and we could skip a not-needed write in some cases.
```suggestion
            # Retrieve the value for a specific invoice
            val_to_updt = updt_val.get(invoice.name)
            if val_to_updt:
                invoice.button_draft()
                # If the invoice is defined to_cancel we update it state
                if "to_cancel" in val_to_updt["invoice_line_ids"]:
                    invoice._cancel_invoice()
                    continue
                else:
                    invoice.write(val_to_updt)
                invoice.action_post()
```